src/main/scala/io/iohk/ethereum/domain/SignedTransaction.scala

```diff
@@ -109,14 +107,13 @@ object SignedTransaction {
 
 case class SignedTransaction (
   tx: Transaction,
-  signature: ECDSASignature,
-  senderAddress: Address) {
+  signature: ECDSASignature) {
```

Comment:
Why not just convert it to a `lazy val`?

Reply:
tbh i was thinking about it, but:
1. Sender field not make sense in all transactions , it would need to be `Option[Sender]` as in wrongly signed transactions we would not be able recover sender, or in case of leaving it as `Sender` I would need to track when exceptions would need to be cought. 
2. I got more help from compiler when removing this field, when converted to lazy val it just compiles more or less as it was, after removing that field i needed to expicilty call `getSender` in each place deciding what to do.